src/main.rs

```diff
@@ -69,130 +67,123 @@ fn select_response(response_type: ResponseType) -> Result<String, String> {
-        let mommys_pronouns = parse_options(MOMMYS_PRONOUNS_ENV_VAR, MOMMYS_PRONOUNS_DEFAULT);
-
-        apply_template(
-            &response,
-            MOMMYS_PRONOUN_PLACEHOLDER,
-            &mommys_pronouns,
-            &rng,
-        )
-    };
+    // Done~!
+    Ok(response)
+}
 
-    let mut response = {
-        const MOMMYS_ROLES_DEFAULT: &str = "mommy";
-        const MOMMYS_ROLES_ENV_VAR: &str = "CARGO_MOMMYS_ROLES";
-        const MOMMYS_ROLE_PLACEHOLDER: &str = "MOMMYS_ROLE";
-        let mommys_roles = parse_options(MOMMYS_ROLES_ENV_VAR, MOMMYS_ROLES_DEFAULT);
-        apply_template(&response, MOMMYS_ROLE_PLACEHOLDER, &mommys_roles, &rng)
-    };
+struct Responses<'a> {
+    vars: &'a [Var<'a>],
+    moods: &'a [Mood<'a>],
+}
 
-    {
-        const MOMMYS_EMOTES_ENV_VAR: &str = "CARGO_MOMMYS_EMOTES";
-        const MOMMYS_EMOTES_DEFAULT: &str = "❤️/💖/💗/💓/💞";
-        let mommys_emotes = parse_options(MOMMYS_EMOTES_ENV_VAR, MOMMYS_EMOTES_DEFAULT);
+include!(concat!(env!("OUT_DIR"), "/responses.rs"));
 
-        let should_emote = rng.bool();
-        if should_emote && !mommys_emotes.is_empty() {
-            let mommys_emotes = parse_options(MOMMYS_EMOTES_ENV_VAR, MOMMYS_EMOTES_DEFAULT);
-            let emote = &mommys_emotes[rng.usize(..mommys_emotes.len())];
-            response.push(' ');
-            response.push_str(emote);
+impl Responses<'_> {
+    fn apply_template(&self, chunks: &[Chunk], rng: &Rng) -> Result<String, String> {
+        let mut out = String::new();
+        for chunk in chunks {
+            match chunk {
+                Chunk::Text(text) => out.push_str(text),
+                Chunk::Var(i) => out.push_str(&self.vars[*i].load(rng)?),
+            }
         }
+        Ok(out)
     }
+}
 
-    // Done~!
-    Ok(response)
+struct Mood<'a> {
+    name: &'a str,
+    positive: &'a [&'a [Chunk<'a>]],
+    negative: &'a [&'a [Chunk<'a>]],
+}
+
+enum Chunk<'a> {
+    Text(&'a str),
+    Var(usize),
 }
 
-fn parse_options(env_var: &str, default: &str) -> Vec<String> {
-    std::env::var(env_var)
-        .unwrap_or_else(|_| default.to_owned())
-        .split('/')
-        .map(|s| s.to_owned())
-        .collect()
+struct Var<'a> {
+    env_key: &'a str,
+    defaults: &'a [&'a str],
 }
 
-fn apply_template(input: &str, template_key: &str, options: &[String], rng: &Rng) -> String {
-    let mut last_position = 0;
-    let mut output = String::new();
-    for (index, matched) in input.match_indices(template_key) {
-        output.push_str(&input[last_position..index]);
-        output.push_str(&options[rng.usize(..options.len())]);
-        last_position = index + matched.len();
+impl Var<'_> {
+    fn load(&self, rng: &Rng) -> Result<String, String> {
+        self.maybe_load(rng).ok_or_else(|| {
+            let role = if std::ptr::eq(self, ROLE) {
+                "mommy(?)".to_string()
+            } else {
+                match ROLE.load(rng) {
```

Comment:
However, this performs a load by relying on pointer equality.

I recently woke up, so I am not 100% awake for the "Rust reference semantics" part of my brain, but this logic is at risk of incorrect reasoning as `ROLE` is a `const`, allowing its de facto duplication and thus spurious pointer inequality.

Even if we could use a `const`, other little ones trying to help mommy might be easily confused, and we wouldn't want that, would we~?